method/DeepST/DeepST.py

```diff
@@ -0,0 +1,227 @@
+    required=False,
+)
+parser.add_argument("--image", help="Path to H&E staining.", required=False)
+parser.add_argument(
+    "--n_clusters", help="Number of clusters to return.", required=True, type=int
+)
+parser.add_argument(
+    "--technology",
+    help="The technology of the dataset (Visium, ST, imaging-based).",
+    required=True,
+)
+parser.add_argument(
+    "--seed", help="Seed to use for random operations.", required=True, type=int
+)
+parser.add_argument(
+    "--config",
+    help="Optional config file (json) used to pass additional parameters.",
+    required=False,
+)
+
+args = parser.parse_args()
+
+from pathlib import Path
+
+out_dir = Path(args.out_dir)
+
+# Output files
+label_file = out_dir / "domains.tsv"
+embedding_file = out_dir / "embedding.tsv"
+# if additional output files are required write it also to out_dir
+
+# Use these filepaths as input ...
+coord_file = args.coordinates
+feature_file = args.features
+observation_file = args.observations
+
+if args.neighbors is not None:
+    neighbors_file = args.neighbors
+if args.matrix is not None:
+    matrix_file = args.matrix
+if args.dim_red is not None:
+    dimred_file = args.dim_red
+if args.image is not None:
+    image_file = args.image
+if args.config is not None:
+    # Def config
+    import json
+    with open(args.config, "r") as f:
+        config = json.load(f)
+
+n_clusters = args.n_clusters
+technology = args.technology
+seed = args.seed
+
+
+# ... or AnnData if you want
+def get_anndata(args):
+    import anndata as ad
+    import numpy as np
+    import pandas as pd
+    import scipy as sp
+    from PIL import Image
+
+    observations = pd.read_table(args.observations, index_col=0)
+    features = pd.read_table(args.features, index_col=0)
+
+    coordinates = (
+        pd.read_table(args.coordinates, index_col=0)
+        .loc[observations.index, :]
+        .to_numpy()
+    )
+
+    adata = ad.AnnData(obs=observations, var=features, obsm={"spatial": coordinates})
+
+    if args.matrix is not None:
+        X = sp.io.mmread(args.matrix)
+        if sp.sparse.issparse(X):
+            X = X.tocsr()
+        adata.X = X
+
+    if args.neighbors is not None:
+        adata.obsp["spatial_connectivities"] = sp.io.mmread(args.neighbors).T.tocsr()
+
+    # Filter by selected samples/features
+    if "selected" in adata.obs.columns:
+        adata = adata[observations["selected"].astype(bool), :]
+    if "selected" in adata.var.columns:
+        adata = adata[:, features["selected"].astype(bool)]
+
+    if args.dim_red is not None:
+        adata.obsm["reduced_dimensions"] = (
+            pd.read_table(args.dim_red, index_col=0).loc[adata.obs_names].to_numpy()
+        )
+
+    if args.image is not None:
+        adata.uns["image"] = np.array(Image.open(args.image))
+    else:
+        adata.uns["image"] = None
+
+    return adata
+
+adata = get_anndata(args)
+adata.var_names_make_unique()
+
+#######
+# NOTE: Seed cannot be used here
```

Comment:
Wondering if we should still try to seed

https://pytorch.org/docs/stable/notes/randomness.html